### DISCUSSION TITLE
Accessing a list of utilities & their CSS counterparts

### DISCUSSION THREAD

**Discussion by JakeAdmire** [tailwindlabs/tailwindcss#14208] [Help]
For a plugin I'm working on, it'd be extremely useful to have access to a JS object that looks something like this:
```js
bg: 'background-color',
text: 'color',
pt: 'padding-top',
// etc
```
Basically just a way to loop over every core utility and see their CSS counterparts

Is there one that exists, or any straightforward way to dynamically create one?

**Comment by wongjn** [comment]
The "mappings" are defined in `tailwind/src/corePlugins.js`:

https://github.com/tailwindlabs/tailwindcss/blob/f07dbff2a7f78fd75c53c6cfe01b58b6c0419f22/src/corePlugins.js#L566-L3007

You could look at `import`ing/`require`ing the `corePlugins` `export` in a JavaScript file and then iterate over the members to create your JavaScript object.

Though do bear in mind, there would be some collisions if we extrapolate from your example. For example:
- The `bg` prefix is used for `background-color`, `background-image`, `background-size`, etc.
- The `text` prefix is used for `color` and `font-size`.
